src/ast/variables/UndefinedVariable.ts

```diff
@@ -0,0 +1,12 @@
+import { LiteralValueOrUnknown } from '../values';
+import Variable from './Variable';
+
+export default class UndefinedVariable extends Variable {
+	constructor() {
+		super('undefined');
+	}
+
+	getLiteralValueAtPath(): LiteralValueOrUnknown {
+		return undefined;
```

Comment:
What cases would this apply to?

Reply:
This applies to all situations where the global `undefined` variable is used.

Reply:
That is all situations where `undefined` is accessed and has not been redeclared.